exec/java-exec/src/main/codegen/templates/ComparisonFunctions.java

```diff
@@ -215,6 +192,36 @@ public void eval() {
     }
   }
 
+  <#-- IS_DISTINCT_FROM function -->
+  @FunctionTemplate(names = {"is_distinct_from", "is distinct from" },
```

Comment:
@vkorukanti, I want to clarify...if the query only had a join condition with IS_NOT_DISTINCT_FROM, I would think it should work just with your convertlet changes, since both HashJoin and MergeJoin handle this type of join condition.  Is the reason you had to implement the full comparator codegen to handle more general types of comparisons ?  e.g in the SELECT list if I say  'SELECT  a IS NOT DISTINCT FROM b'  ?    Suppose we had a convertlet that only preserved the IS (NOT) DISTINCT FROM join condition, and defaulted to the Calcite rewrite using the CASE expression, then we would not have to implement the full comparator.